### PR TITLE
feat(api): add `concat` and `repeat` methods for easier array operation discovery

### LINKED ISSUE #6720: feat: document Array.__add__()
### Is your feature request related to a problem?

I was looking for Array.concat(), and couldn't find it. I had to go try `a + b`, and that worked, but that would be nice if it was discoverable.

### Describe the solution you'd like

Add __add__() to the docs. OR perhaps just add in an explicit `.concat()` method to Arrays so it is even easier. At least in the __add__method docstring include the keyword concat so that it shows up in search.

### What version of ibis are you running?

main

### What backend(s) are you using, if any?

Na

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by cpcloud**
Make array concatenation and repetition easier to discover.

Closes #6720.

### CODE REVIEW

File: ibis/expr/types/arrays.py [hunk 1]
```diff
@@ -153,71 +155,59 @@ def __add__(self, other: ArrayValue) -> ArrayValue:
...
         └──────────────────────┘
         """
-        return ops.ArrayConcat(other, self).to_expr()
+        return ops.ArrayConcat((self, other, *args)).to_expr()
+
+    def __add__(self, other: ArrayValue) -> ArrayValue:
+        return self.concat(other)
+
+    def __radd__(self, other: ArrayValue) -> ArrayValue:
+        return ops.ArrayConcat((other, self)).to_expr()
```
Comment (kszucs) [inline ibis/expr/types/arrays.py:208]: `other.concat(self)`?

**Review by kszucs — APPROVED** [review]
Nice!

File: ibis/expr/types/arrays.py (hunk 1 above)
Comment (cpcloud) [reply ibis/expr/types/arrays.py:208]: `other` can be a different type than `ArrayValue`. I should probably remove or fix the type annotation there.

Comment (kszucs) [reply ibis/expr/types/arrays.py:208]: Right, and `self` cannot be used because of the concat order.

Comment (kszucs) [reply ibis/expr/types/arrays.py:208]: Regarding the annotations we should have type aliases at some point, e.g. `ArrayLike` which contains the types coercible to array values as well.

**Pull request merged**